pkg/kds/global/components.go

```diff
@@ -115,16 +115,30 @@ func Setup(rt runtime.Runtime) error {
 	onGlobalToZoneSyncConnect := mux.OnGlobalToZoneSyncConnectFunc(func(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer, errChan chan error) {
 		zoneID, err := util.ClientIDFromIncomingCtx(stream.Context())
 		if err != nil {
-			errChan <- err
+			select {
+			case errChan <- err:
+			default:
+			}
+			return
 		}
 		log := kdsDeltaGlobalLog.WithValues("peer-id", zoneID)
 		log = kuma_log.AddFieldsFromCtx(log, stream.Context(), rt.Extensions())
 		log.Info("Global To Zone new session created")
 		if err := createZoneIfAbsent(stream.Context(), log, zoneID, rt.ResourceManager()); err != nil {
-			errChan <- errors.Wrap(err, "Global CP could not create a zone")
+			select {
+			case errChan <- err:
+			default:
+				log.Error(errors.Wrap(err, "Global CP could not create a zone"), "failed to write error to closed channel")
```

Comment:
shouldn't the wrapped error be sent to the channel?

Reply:
correct, forgot to change it